lib/pokerscore/pair_search.rb

```diff
@@ -0,0 +1,16 @@
+require 'lib/pokerscore/set_factories'
+
+module PokerScore
+  class PairSearch
+
+    def call(hand)
+      pairs = []
+      grouped_cards_by_value = hand.cards.group_by(&:value)
+      grouped_cards_by_value.each_value do |cards|
+        pairs << PokerScore::SetFactories::pair(cards) if cards.size == 2
+      end
+      pairs
+    end
```

Comment:
We might be able to get away with dropping the `grouped_cards_by_value` variable here?
```
hand.cards.group_by(&:value) do |cards_with_same_value|
  pairs << PokerScore::SetFactories::pair(cards_with_same_value) if cards.size == 2
end
```

We _could_ also get fancy with a `tap` here, though last time I did this my team members were up in arms (though I still think it's worth using, as I don't like having to explicitly return `pairs`)

```
def call(hand)
  [].tap do |pairs|
    hand.cards.group_by(&:value) do |cards_with_same_value|
      pairs << PokerScore::SetFactories::pair(cards_with_same_value) if cards.size == 2
    end
  end
end
```

This is mostly syntactic sugar though, and is somewhat contentious (at least to Dennis and Jon)

Reply:
Addressed in person, and has been cleaned up